### PR TITLE
blivet-gui doesn't handle known StorageError,so we see crash on situations like bug#1756288.It seems that we should provide users a message to let them change their behaviors rather than crash the installation.

### PR DESCRIPTION
**Pull request by skycastlelily2**

### CODE REVIEW

**Comment by skycastlelily2** [comment]
Hi Vojtech,
  Thanks a lot for your feedback.Do you want me to override the _reraise_exception with something like the following(just a draft):
 def _reraise_exception(self, exception, traceback):
        from .gui_utils import command_exists
        allow_report = command_exists("gnome-abrt")
        allow_ignore =  not issubclass(type(exception), errors.CommunicationError)
        if allow_ignore:
            msg = _("Unknown error occured.\n{error}").format(error=str(exception))
        else:
            msg = _("Unknown error occured. Blivet-gui will be terminated.\n{error}").format(error=str(exception))

        dialog = message_dialogs.ExceptionDialog(self.main_window, allow_ignore,
                                                 allow_report, msg, str(traceback))
        response = dialog.run()

        if response == constants.DialogResponseType.BACK:
            return
        else:
            if response == constants.DialogResponseType.REPORT:
               subprocess.call(["gnome-abrt"])
  
 If so,I would have some concerns: 
 1) It seems that we can't get the allow_ignore value in osinstall.BlivetGUIAnaconda class,or please help me out by telling me how to get the value,thanks in advance. Can we remove the allow_ignore option ?Users can continue the installation without click "Back" button.
2) You know that abrt daemon won't be started in anaconda environment even if you run "gnome-abrt",right? So users won't be able to report a bug during the installation even if they click the "report a bug" button.

**Comment by skycastlelily2** [comment]
Hi Vojtech,
     Sorry about making noise before look deep into the code,as I said in the last comment,starting
gnome-abrt won't let user report bug,as the daemons aren't running in anaconda environment.
I'm thinking maybe we can let anaconda to do the magic if users really want to report a bug.

**Comment by vojtechtrefny** [comment]
> Sorry about making noise before look deep into the code,as I said in the last comment,starting
gnome-abrt won't let user report bug,as the daemons aren't running in anaconda environment.
I'm thinking maybe we can let anaconda to do the magic if users really want to report a bug

Sorry I didn't anwer sooner. Yes, this is the idea had in mind -- just re-raising the exception if user wants to report the bug so Anaconda will catch it and run python-meh to deal with it.

**Comment by skycastlelily2** [comment]
Nope,really my pleasure,and thanks a lot for your time on guiding me.

File: blivetgui/osinstall.py [hunk 1]
```diff
@@ -198,6 +200,24 @@ def enlightbox(self):
...
+        dialog = message_dialogs.ExceptionDialog(self.main_window, allow_ignore,
+                                                 allow_report, msg, str(traceback))
+        response = dialog.run()
+
+        if response == constants.DialogResponseType.BACK:
+            return
+        else:
+            if response == constants.DialogResponseType.REPORT:
+                raise type(exception)(message + str(exception) + "\n" + traceback)
+
```
Comment (vojtechtrefny) [inline blivetgui/osinstall.py]: I didn't realize this before, but we need to handle the QUIT response here too. Right now it's basically same as BACK. Quit option in python-meh just calls `sys.exit` -- https://github.com/rhinstaller/python-meh/blob/master/meh/handler.py#L184 -- so I guess blivet-gui should do the same.

**Comment by vojtechtrefny** [comment]
I've tested this manually in anaconda and found one small issue, but other than that everything seems to be working as expected.

Also pycodestyle is now complaining about some trailing whitespaces
```
blivetgui/blivetgui.py:279:52: W291 trailing whitespace
blivetgui/blivetgui.py:287:1: W293 blank line contains whitespace
blivetgui/blivetgui.py:312:1: W293 blank line contains whitespace
blivetgui/blivetgui.py:564:1: W293 blank line contains whitespace
blivetgui/blivetgui.py:565:18: W291 trailing whitespace
```

**Comment by skycastlelily2** [comment]
Sorry for not realizing that beforehand,and really thanks for your patience on polishing this PR,
 will try to consider and test thoroughly before send PRs in the future^^

**Comment by skycastlelily2** [comment]
Sorry for not realizing that beforehand,and really thanks for your patience on polishing this PR,
 will try to consider and test thoroughly before send PRs in the future^^

**Comment by skycastlelily2** [comment]
Sorry for not realizing that beforehand,and really thanks for your patience on polishing this PR,
will try to consider and test thoroughly and do pylint before send PRs in the future^^

**Pull request merged**